### ISSUE TITLE
Problem with backpropagation when combining Reshape() and Padding()

### ISSUE THREAD

**Issue by tlemberger** [torch/nn#1276]
When working on nested sequencers in the context of an encoder-decoder architecture, I ran into problems with backpropagation (similar issue as https://github.com/Element-Research/rnn/issues/329). Irrespective of the nested sequencers context, it seems that (one of) the problem(s) may boil down to the combined used of `nn.Reshape()` (or alternatively `nn.Unsqueeze()`) with `nn.Padding()`.

Here is a minimal toy example:

```
require 'nn'
m = nn.Sequential()--in: 3 x 2 
m:add(nn.Reshape(1, 2, true))--out: 3 x 1 x 2
m:add(nn.Padding(2, 9))--out: 3 x 10 x 2
c = nn.MSECriterion()
x = torch.ones(3,2)
m:zeroGradParameters()
y=m:forward(x)
c:forward(y,y)
g=c:backward(y,y)
m:listModules()
m:backward(y,g)
```
raises this error

```
In 1 module of nn.Sequential:
...torch/install/share/lua/5.1/torch/Tensor.lua:466: Wrong size for view. Input size: 3x1x2. Output size: 3x10x2
```	
Using `nn.Padding()` alone

```
m = nn.Sequential()--in: 3 x 1 x 2 
m:add(nn.Padding(2, 9))--out: 3 x 10 x 2
c = nn.MSECriterion()
x = torch.ones(3,1,2)
m:zeroGradParameters()
y=m:forward(x)
c:forward(y,y)
g=c:backward(y,y)
m:listModules()
m:backward(y,g)
```
or  `nn.Reshape()` alone

```
m = nn.Sequential()--in: 3 x 2 
m:add(nn.Reshape(1, 2, true))--out: 3 x 1 x 2
c = nn.MSECriterion()
x = torch.ones(3,2)
m:zeroGradParameters()
y=m:forward(x)
c:forward(y,y)
g=c:backward(y,y)
m:listModules()
m:backward(y,g)
```
works even if the input and output shapes are different.

It is only the combination of both that fails. Using `nn.Padding()` first followed by `nn.Reshape()` has the same problem.

Am I doing something wrong?
